lib/kafo/exit_handler.rb

```diff
@@ -1,11 +1,10 @@
 module Kafo
   class ExitHandler
-    attr_accessor :cleanup_paths, :exit_code, :logger
+    attr_accessor :cleanup_paths, :exit_code
```

Comment:
it may be better to keep logger as a writer and add customer reader

```
def logger
  @logger ||= KafoConfigure.logger
end
```
this way we also don't require KafoConfigure to be ready during initialization but with the first use of local logger

the benefit is, logger can still be changed to another instance if needed

Reply:
I also considered that, but was confused by https://github.com/theforeman/kafo/blob/afc25fe69570db9dd6ad8abd232e7bdfde359ebb/lib/kafo/exit_handler.rb#L30-L31

Should that also use the logger?

Reply:
probably yes

Reply:
My issue is that this is all highly inconsistent:
https://github.com/theforeman/kafo/blob/afc25fe69570db9dd6ad8abd232e7bdfde359ebb/lib/kafo/hooking.rb#L22-L24
https://github.com/theforeman/kafo/blob/afc25fe69570db9dd6ad8abd232e7bdfde359ebb/lib/kafo/migrations.rb#L23
https://github.com/theforeman/kafo/blob/afc25fe69570db9dd6ad8abd232e7bdfde359ebb/lib/kafo/puppet_module.rb#L16
https://github.com/theforeman/kafo/blob/afc25fe69570db9dd6ad8abd232e7bdfde359ebb/lib/kafo/puppet_module.rb#L45

For now I'll just fix this class.